velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -268,6 +268,25 @@ TEST_F(ArithmeticTest, cosh) {
   EXPECT_TRUE(std::isnan(cosh(kNan).value_or(0)));
 }
 
+TEST_F(ArithmeticTest, unhex) {
+  const auto unhex = [&](std::optional<std::string> a) {
+    return evaluateOnce<std::string>("unhex(c0)", a);
+  };
+
+  EXPECT_EQ(unhex("737472696E67"), "string");
+  EXPECT_EQ(unhex(""), "");
+  EXPECT_EQ(unhex("23"), "#");
+  std::string b("#\0", 2);
+  EXPECT_EQ(unhex("123"), b);
+  EXPECT_EQ(unhex("b23"), b);
+  b = std::string("##\0", 3);
+  EXPECT_EQ(unhex("b2323"), b);
+  EXPECT_EQ(unhex("F"), "\x0F");
+  EXPECT_EQ(unhex("ff"), "\xFF");
+  EXPECT_EQ(unhex("GG"), std::nullopt);
```

Comment:
Would you add a test with a single invalid character as well as single invalid character followed by even number of valid characters?

Reply:
add two tests
```
  EXPECT_EQ(unhex("G"), std::nullopt);
  EXPECT_EQ(unhex("G23"), std::nullopt);
```